Lib/Protocols/IdSSLOpenSSL.pas

```diff
@@ -2412,6 +2423,17 @@ procedure UnLoadOpenSSLLibrary;
   LList: TIdCriticalSectionList;
 {$ENDIF}
 begin
+  //
+  // Zdravko Gabrovski - check for GLOBAL OpenSSL handlers
+  //
+  {$IFNDEF STATICLOAD_OPENSSL}
+    if ( GlobalIdCrypto <> IdNilHandle ) and ( GlobalIdSSL <> IdNilHandle ) then begin
+      InitializeFuncPointers;
+      SSLIsLoaded.Value := False;
+      Exit;
+    end;
+  {$ENDIF}
+

```

Comment:
If the 2 global handles have been assigned, exiting here will bypass Indy's cleanup of OpenSSL's thread-locking callbacks. Are you sure this is intended behavior?